core/src/main/java/org/apache/accumulo/core/clientImpl/ScannerIterator.java

```diff
@@ -112,16 +120,32 @@ public boolean hasNext() {
     throw new NoSuchElementException();
   }
 
+  void close() {
+    // run actual close operation in the background so this does not block.
+    readaheadPool.execute(() -> {
+      synchronized (scanState) {
```

Comment:
Is it possible to consume all the threads in the pool while waiting for ThriftScanner to close?  Like if I were to call close in a loop.

Reply:
No,  I think that pool will created an unlimited number of threads.  Your question made me realize I was probably tracking a max of too many iterators.  I lowered it from 1000 to 32.  The use case I am trying to support in someone creating a scanner, creating a few iterators, and then closing.  If someone is created a large number of iterators, I am not really trying to support that.

Reply:
I am thinking ideally the close messages would be grouped by tablets server and then the messages sent.  

This makes me think it would be nice to have a service for making async calls.  I can just queue up async calls I want to make and the service will group them by tablet server.